shared/js/l10n.js

```diff
@@ -1172,10 +1175,22 @@
 
       switch (type) {
         case 'json':
-          io.loadJSON(path, onJSONLoaded, sync);
+          if (source === 'extra') {
+            navigator.mozApps.getLocalizationResource(
+              this.id, gaiaVersion, path, 'json')
+                .then(onJSONLoaded.bind(this, null), onL10nLoaded);
+          } else {
+            io.loadJSON(path, onJSONLoaded, sync);
+          }
           break;
         case 'properties':
-          io.load(path, onPropLoaded, sync);
+          if (source === 'extra') {
```

Comment:
I like it.

I even started thinking about maybe moving this to io? This way IO would start being part of the bindings where it really belongs and we'd have langpacks be an artifact of the environment with library intact.
I'm ok with leaving this for 3.0 because that would require io to manage localeSources and know which language it is trying to load, which right now belongs to Locale.

What do you think?